### PR TITLE
Armfido script fixes

### PR DESCRIPTION
**Pull request by KyleAnthonyShepherd**
brought logic more in line with other "trajectory change" weapons
adjusted heading checks to be in line with the 3 frame delay required by wait-for-turn
improved signal handling, so AimWeapon and Activate/Deactivate signal kill aim animations/turns, stopping barrel wiggling
change weapon state directly in the Activate/Deactivate calls, removing unncessary RequestState handing function
removed Reload function, reload time is now controlled in the unitdef
unitdef reload time of gauss changed to match old in-game value of 82 frame reload time, to preserve existing balance

Technically, this change is still a tiny buff to armfido, as now gauss reload time will decrease with XP gain.

### CODE REVIEW

File: scripts/Units/armfido.bos [hunk 1]
```diff
@@ -6,13 +6,14 @@
 piece  barrel, turret, loturret, flare, sleeve, torso, pelvis, lthigh, lleg, lfoot, rthigh, rleg, rfoot, luparm, lloarm, lhand, ruparm, rloarm, rhand,
        eyes, shoulder, ltooth, rtooth, aimy1, aimx1;
 
-static-var  restore_delay, Static_Var_2, gun_2, statechg_DesiredState,
-			statechg_StateChanging, moveSpeed, currentSpeed, wpn1_lasthead, wpn2_lasthead, animSpeed, maxSpeed, animFramesPerKeyframe, bMoving;
+static-var  restore_delay, Static_Var_2, moveSpeed, currentSpeed, animSpeed, maxSpeed, animFramesPerKeyframe, bMoving;
```
Comment (WatchTheFort) [inline scripts/Units/armfido.bos:9]: `Static_Var_2` -> `currentWeapon`, and add constants for the weapons, e.g. `#define GAUSS 0`, then the functionality is clear in the code, and explanatory comments aren't needed.
`if (currentWeapon == GAUSS) { }`

File: scripts/Units/armfido.bos [hunk 2]
```diff
@@ -6,13 +6,14 @@
...
+
+static-var last_primary_heading,last_primary_pitch;
 
 // Signal definitions
 #define SIG_AIM				2
 #define SIG_AIM_2			8
```
Comment (WatchTheFort) [inline scripts/Units/armfido.bos:15]: `SIG_AIM_2` is unused and can be deleted

File: scripts/Units/armfido.bos [hunk 3]
```diff
@@ -352,21 +327,33 @@ Create()
...
 	moveSpeed = get MAX_SPEED;
+	// set last heading values to large negative, ensure first aim waits for the turn
+	last_primary_heading = -1000000;
+	last_primary_pitch = -1000000;
 }
 
 Activate()
 {
-	start-script RequestState(0);
+	// signal kills any current aim animations
```
Comment (WatchTheFort) [inline scripts/Units/armfido.bos:339]: Comments explaining `signal` and `set-signal-mask` aren't generally needed, as this is base BOS functionality

File: scripts/Units/armfido.bos [hunk 4]
```diff
@@ -317,32 +318,6 @@ StopMoving(){
 
 #include "unit_hitbyweaponid_and_smoke.h"
 
-RequestState(requestedstate, currentstate)
```
Comment (WatchTheFort) [inline scripts/Units/armfido.bos:320]: Does this mean the other custom on/off unit scripts don't need the `RequestState` function? I was under the impression it was necessary for on/off functionality, as it is a built-in script function defined by the engine (see https://units.tauniverse.com/tutorials/tadesign/tadesign/bosdesc.htm).

**Review by WatchTheFort — CHANGES_REQUESTED** [review]

File: scripts/Units/armfido.bos (hunk 1 above)
Comment (KyleAnthonyShepherd) [reply scripts/Units/armfido.bos:9]: That makes sense, will do

File: scripts/Units/armfido.bos (hunk 3 above)
Comment (KyleAnthonyShepherd) [reply scripts/Units/armfido.bos:339]: It serves to explain why signal is used here, so it is not removed in the future by accident. When armfido.bos was transitioned for unit_continuous_aim.lua compatibility, signals were stripped out leading to animation errors, two or more turns trying to complete.

File: scripts/Units/armfido.bos (hunk 4 above)
Comment (KyleAnthonyShepherd) [reply scripts/Units/armfido.bos:320]: https://github.com/beyond-all-reason/Beyond-All-Reason/blob/master/scripts/statechg.h
It seems to be defined by the game and not the engine.
I think RequestState is better used for units that have a distinct 3rd state, a "changing from state A to state B" state. armfido switch is instant, so RequestState is unnecessary.

**Comment by Beherith** [comment]
In we go, thanks!

**Pull request merged**